### ISSUE TITLE
Refactor make-generic-rpm to support Debian

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#1117]
Probably need to rename it, too...

**Comment by mfeit-internet2** [comment]
This has turned into a huge project.  :-)

The big stumbling block right now is getting `pscheduler-server` to do the syntax check during build and install properly.  Think I'm going to wait until Antoine is back from vacation to tackle that.  The rest is plugins, which should be smooth sailing.

Known TODOs:

```
./python-attrs/debian/control:### TODO: The presence of these packages causes _nothing_ to be packaged.
./python-attrs/debian/rules:# TODO: Make this figure this out programmatically.
./make-generic-package/make-generic-package/debian/changelog:  * DUMMY VERSION.  TODO: Remove this.
./s3-benchmark/debian/rules:# TODO: This fails to build.
./pscheduler-server/pscheduler-server/debian/patches/reset.patch: # TODO: This will have to be a separate program once the database
./pscheduler-server/pscheduler-server/debian/rules:# TODO: The database build needs to check syntax, which requires Pg as
```

**Comment by mfeit-internet2** [comment]
Tabling until 5.1.

**Comment by mfeit-internet2** [comment]
Known TODOs:

- [ ] Make sure patches aren't duplicated in Debian directories

**Comment by mfeit-internet2** [comment]
Holding this for 5.1.  No critical need to do it for 5.0.

**Comment by tonin** [comment]
Not sure if this is a Debian build or packaging related error or a more generic one, but with the current state of branch issue-1117 I see this error when running the `pscheduler-server` unit tests:

```
Limit Processor / Limit PassFail / Data Validation ... ok
Limit Processor / Limit PassFail / Limit ... ok
Limit Processor / Limit URLFetch / Data Validation ... ok
Limit Processor / Limit URLFetch / Limit ... ok
Logging tests ... --- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.7/logging/handlers.py", line 933, in emit
    self.socket.send(msg)
OSError: [Errno 9] Bad file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/logging/handlers.py", line 854, in _connect_unixsocket
    self.socket.connect(address)
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.7/logging/handlers.py", line 936, in emit
    self._connect_unixsocket(self.address)
  File "/usr/lib/python3.7/logging/handlers.py", line 865, in _connect_unixsocket
    self.socket.connect(address)
FileNotFoundError: [Errno 2] No such file or directory
Call stack:
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3/dist-packages/nose/__main__.py", line 8, in <module>
    run_exit()
  File "/usr/lib/python3/dist-packages/nose/core.py", line 121, in __init__
    **extra_args)
  File "/usr/lib/python3.7/unittest/main.py", line 101, in __init__
    self.runTests()
  File "/usr/lib/python3/dist-packages/nose/core.py", line 207, in runTests
    result = self.testRunner.run(self.test)
  File "/usr/lib/python3/dist-packages/nose/core.py", line 62, in run
    test(result)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3.7/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 75, in run
    test(result)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 177, in __call__
    return self.run(*arg, **kw)
  File "/usr/lib/python3/dist-packages/nose/suite.py", line 224, in run
    test(orig)
  File "/usr/lib/python3/dist-packages/nose/case.py", line 45, in __call__
    return self.run(*arg, **kwarg)
  File "/usr/lib/python3/dist-packages/nose/case.py", line 133, in run
    self.runTest(result)
  File "/usr/lib/python3/dist-packages/nose/case.py", line 151, in runTest
    test(result)
  File "/usr/lib/python3.7/unittest/case.py", line 663, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python3.7/unittest/case.py", line 615, in run
    testMethod()
  File "/app/python-pscheduler/unibuild-work/build/python-pscheduler/.pybuild/cpython3_3.7_pscheduler/build/tests/log_test.py", line 27, in test_log
    log = Log(verbose=False, prefix='test')
  File "/app/python-pscheduler/unibuild-work/build/python-pscheduler/.pybuild/cpython3_3.7_pscheduler/build/pscheduler/log.py", line 233, in __init__
    self.info("Started")
  File "/app/python-pscheduler/unibuild-work/build/python-pscheduler/.pybuild/cpython3_3.7_pscheduler/build/pscheduler/log.py", line 340, in info
    self.__log(self.logger, INFO, format, *args)
  File "/app/python-pscheduler/unibuild-work/build/python-pscheduler/.pybuild/cpython3_3.7_pscheduler/build/pscheduler/log.py", line 326, in __log
    logger.log(level, line)
Message: 'Started'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/app/python-pscheduler/unibuild-work/build/python-pscheduler/.pybuild/cpython3_3.7_pscheduler/build/tests/log_test.py", line 32, in test_log
    raise ValueError("Test exception")
ValueError: Test exception

```

**Comment by mfeit-internet2** [comment]
Per discussion:

- [ ] Add syslog to build requirements on Debian (@tonin)
- [ ] Do the same for RH (Me).

Make sure the service is started before running tests.

**Comment by mfeit-internet2** [comment]
This was merged into 5.0.0.   No reason to keep it open.
